### PR TITLE
[FIX] Travis build for macOS. Also add cmake build test.

### PR DESCRIPTION
**Pull request by saurabhshri**
**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/ccextractor/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.
- ~[ ] I have mentioned this change in the [changelog]~

**My familiarity with the project is as follows (check one):**

- [ ] I have never used CCExtractor.
- [ ] I have used CCExtractor just a couple of times.
- [ ] I absolutely love CCExtractor, but have not contributed previously.
- [x] I am an active contributor to CCExtractor.

---

- Fixed failing Travis build for macOS. Also added cmake build test in the same.
- Removed linux build tests because Sample Platform already covers it.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,101 @@
...
-before_install:
-- if [[ $TRAVIS_OS_NAME == 'osx' ]]; then brew upgrade automake; brew install pkg-config autoconf automake libtool tesseract leptonica; fi
-- if [[ $TRAVIS_OS_NAME == 'linux' ]]; then sudo apt-get install -y libcurl4-gnutls-dev tesseract-ocr tesseract-ocr-dev libleptonica-dev autoconf-archive; fi
-- if [[ $TRAVIS_OS_NAME == 'linux' ]]; then sudo apt-get install python-distutils-extra tesseract-ocr tesseract-ocr-eng libopencv-dev libtesseract-dev libleptonica-dev python-all-dev swig libcv-dev python-opencv python-numpy python-setuptools build-essential subversion; fi
-- if [[ $TRAVIS_OS_NAME == 'linux' ]]; then wget https://github.com/DanBloomberg/leptonica/releases/download/1.74.4/leptonica-1.74.4.tar.gz && tar xvf leptonica-1.74.4.tar.gz; cd leptonica-1.74.4; ./configure && make && sudo make install; cd ..; fi
-- if [[ $TRAVIS_OS_NAME == 'linux' ]]; then git clone https://github.com/tesseract-ocr/tesseract.git; cd tesseract; ./autogen.sh; ./configure --enable-debug; LDFLAGS="-L/usr/local/lib" CFLAGS="-I/usr/local/include" make; sudo make install; sudo ldconfig; cd ..; fi
+    - os: osx
+      osx_image: xcode10.1
+      compiler: clang
+      addons:
```
Comment (canihavesomecoffee) [inline .travis.yml]: It looks like you have a lot of whitespace after the end of line... Care to correct? :)

Comment (saurabhshri) [reply .travis.yml]: Hi! I would love to correct it, but am unable to locate the extra whitespaces. How do I fix it?

screenshot 2019-01-12 at 7 38 02 pm

Comment (canihavesomecoffee) [reply .travis.yml]: There's none -_-

It's because the line on the left side is wrapping over multiple ones that this side is as well... My bad.

Comment (saurabhshri) [reply .travis.yml]: Ah, let me know if any changes are needed 🙂

**Pull request merged**